past_image:
  general:
    eval_method: Spirit  # Default is Spirit.
    image_queue_length: 30  # Use None for infinite.
  Spirit:
    coeff_centrality: 0
    coeff_centrality2: 4
    coeff_direction: 2
    coeff_distance: 8
    coeff_direction_with_current: 2
    coeff_distance_with_current: 1
    ref_distance: 2.5  # m
    thresh_distance: 0.25  # m
    thresh_yaw: 10  # deg
  ConstantDistance:
    coeff_distance: 1
    ref_distance: 1.5
  ConstantTimeDelay:
    coeff_time: 1
    ref_delay: 2
  Murata:
    coeff_height: 2
    coeff_direction: 0.4
    coeff_elevation: 0.8
    coeff_distance: 4
    ref_height: 0.3
    ref_distance: 1.5

drone:
  ip: 192.168.1.1  # Default is 192.168.1.1.
  no_hull: false
  outdoor_hull: false
  loop_rate: 200  # Hz
  altitude_max: 2000  # mm
  altitude_min: 0  # mm
  max_body_angle: 5  # deg
  control_vz_max: 700  # mm/s
  control_yaw: 90  # deg/s

control:
  use_joystick: true  # Default is true.
  js_number: auto  # Default is 1. /dev/input/js1
  button:
    land: 8
    arrived: 9
    takeoff: 10
    emergency: 11
  axis:
    roll: 0
    pitch: 1
    yaw: 2
    z: 3
  scale:
    roll: 1
    pitch: 1
    yaw: 1
    z: 1

pose:
  mock_pose: false  # Default is false.
  real:  # Only used if mock_pose is false
    use_mocap: true  # Default is true.
    use_odometry: false  # Default is false.

camera:
  mock_camera: false  # Default is false.
  show: true  # Default is false.
  streams:
    realtime: false
    slow: true
    past_image: false
  slowdown: 15  # times. Default is 15.

visualization:
  show: true
  debug: false  # false, online, offline
